FinSearchUnified/BusinessLogic/ExportService.php

```diff
@@ -0,0 +1,208 @@
+            $this->errors['general'][] = 'No article found with given ID';
+            return null;
+        }
+
+        return $shopwareArticles;
+    }
+
+    public function generateFindologicProducts($shopwareArticles, $logErrors = false)
+    {
+        $findologicArticles = [];
+
+        try {
+            /** @var Article $article */
+            foreach ($shopwareArticles as $article) {
+                $findologicArticle = self::getFindologicArticle($article, $logErrors);
+
+                if ($findologicArticle) {
+                    $findologicArticles[] = $findologicArticle;
+                }
+            }
+        } catch (Exception $e) {
+            die($e->getMessage());
+        }
+
+        return $findologicArticles;
+    }
+
+    /**
+     * @param Article $shopwareArticle
+     * @param bool $logErrors
+     * @return null|Item
+     * @throws Exception
+     */
+    public function getFindologicArticle($shopwareArticle, $logErrors)
+    {
+        $articleId = $shopwareArticle->getId();
+        $inactiveCatCount = 0;
+        $totalCatCount = 0;
+
+        /** @var Category $category */
+        foreach ($shopwareArticle->getCategories() as $category) {
+            if (!$category->isChildOf($this->baseCategory)) {
+                continue;
+            }
+
+            if (!$category->getActive()) {
+                $inactiveCatCount++;
+            }
+
+            $totalCatCount++;
+        }
+
+        if (!$shopwareArticle->getActive()) {
+            if ($logErrors) {
+                $this->errors[$articleId][] = 'Product is not active';
+            }
+            return null;
+        }
+
+        if ($totalCatCount === $inactiveCatCount) {
+            if ($logErrors) {
+                $this->errors[$articleId][] = sprintf('All %d categories are inactive', $totalCatCount);
+            }
+            return null;
+        }
+
+        try {
+            if ($shopwareArticle->getMainDetail() === null || !$shopwareArticle->getMainDetail()->getActive()) {
+                if ($logErrors) {
+                    $this->errors[$articleId][] = 'Main Detail is not active or not available';
+                }
+                return null;
+            }
+        } catch (EntityNotFoundException $exception) {
+            if ($logErrors) {
+                $this->errors[$articleId][] = sprintf('EntityNotFoundException: %s', $exception->getMessage());
+            }
+            return null;
+        }
+
+        try {
+            $findologicArticle = $this->findologicArticleFactory->create(
+                $shopwareArticle,
+                $this->shopKey,
+                $this->allUserGroups,
+                [],
+                $this->baseCategory
+            );
+
+            if ($findologicArticle->shouldBeExported) {
+                return $findologicArticle->getXmlRepresentation();
+            } elseif ($logErrors) {
+                $this->errors[$articleId][] = 'shouldBeExported is false';
```

Comment:
I think that case is pretty common. For Shopware 6 we are were precise but I do not know how hard we want to go here. That's up to you.

Reply:
I think I will let it here, because it does no harm and only provides additional feedback